presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeometryType.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.AbstractVariableWidthType;
+import com.facebook.presto.spi.type.TypeSignature;
+import io.airlift.slice.Slice;
+
+import static java.lang.String.format;
+
+public class GeometryType
+        extends AbstractVariableWidthType
+{
+    public static final GeometryType GEOMETRY = new GeometryType();
+    public static final String NAME = "Geometry";
+
+    private GeometryType()
+    {
+        super(new TypeSignature(NAME), Slice.class);
+    }
+
+    protected GeometryType(TypeSignature signature)
+    {
+        super(signature, Slice.class);
+    }
+
+    @Override
+    public void appendTo(Block block, int position, BlockBuilder blockBuilder)
+    {
+        if (block.isNull(position)) {
+            blockBuilder.appendNull();
+        }
+        else {
+            block.writeBytesTo(position, 0, block.getSliceLength(position), blockBuilder);
+            blockBuilder.closeEntry();
+        }
+    }
+
+    @Override
+    public Slice getSlice(Block block, int position)
+    {
+        return block.getSlice(position, 0, block.getSliceLength(position));
+    }
+
+    @Override
+    public void writeSlice(BlockBuilder blockBuilder, Slice value)
+    {
+        writeSlice(blockBuilder, value, 0, value.length());
+    }
+
+    @Override
+    public void writeSlice(BlockBuilder blockBuilder, Slice value, int offset, int length)
+    {
+        blockBuilder.writeBytes(value, offset, length).closeEntry();
+    }
+
+    @Override
+    public Object getObjectValue(ConnectorSession session, Block block, int position)
+    {
+        if (block.isNull(position)) {
+            return null;
+        }
+
+        return format("<%s>", getTypeSignature()).getBytes();
```

Comment:
While there isn't a strict contract for what `getObjectValue` should return, it's a good idea to return a json-serializable object that represents the value. This is currently used by the client protocol and can be handy for tests.

Reply:
I am not very clear about it, is it good to return typeSignature?

Reply:
It might be better to return the WKT for the value, if possible. This is what's going to be rendered by the CLI

Reply:
will update